services/ceo_ledger.py

```
@@ -0,0 +1,493 @@
+                    .sort("last_progress_at", 1)
+                    .limit(limit)
+                )
+            except Exception as exc:  # noqa: BLE001
+                log.warning("CEOLedger.open_goals failed: %s", exc)
+                return []
+        elif self._mode == "sqlite" and self._sqlite is not None:
+            try:
+                with self._lock:
+                    cur = self._sqlite.execute(
+                        f"SELECT doc FROM {_TBL} WHERE state NOT IN ('closed','abandoned') "  # nosec B608 — _TBL is a constant
+                        "ORDER BY updated_at ASC LIMIT ?",
```

Ruff (0.15.21)

[error] Possible SQL injection vector through string-based query construction

(S608)